src/lib.rs

```diff
@@ -484,7 +563,48 @@ impl Printer for PrintlnPrinter {
 
 struct NullLoader;
 impl ModuleScriptLoader for NullLoader {
-    fn load_script(&mut self, _: String) -> Option<String> { None }
+    fn load_script(&mut self, _: String) -> Option<String> {
+        None
+    }
+}
+
+pub struct BasicFileLoader {
+    base_dir: PathBuf,
+}
+impl BasicFileLoader {
+    pub fn new() -> BasicFileLoader {
+        BasicFileLoader {
+            base_dir: ".".into(),
+        }
+    }
+    pub fn base_dir<P: AsRef<Path>>(&mut self, base_dir: P) -> &mut Self {
```

Comment:
```suggestion
    pub fn base_dir<P: AsRef<Path>>(self, base_dir: P) -> Self {
```

I don't see a point in making it take `&mut self` because it will eventually be owned by the VM anyways.